examples/src/bin/orientable_subclass.rs

```diff
@@ -0,0 +1,176 @@
+                "Orientation",
+                gtk::Orientation::static_type(),
+                gtk::Orientation::Horizontal.to_glib(),
+                glib::ParamFlags::READWRITE | glib::ParamFlags::CONSTRUCT,
+            )
+        })];
+
+    impl ObjectSubclass for CustomOrientable {
+        const NAME: &'static str = "ExCustomOrientable";
+        type Type = super::CustomOrientable;
+        type ParentType = gtk::Widget;
+        type Instance = glib::subclass::simple::InstanceStruct<Self>;
+        type Class = glib::subclass::simple::ClassStruct<Self>;
+
+        glib::object_subclass!();
+
+        fn type_init(type_: &mut glib::subclass::InitializingType<Self>) {
+            type_.add_interface::<gtk::Orientable>();
+        }
+
+        fn class_init(klass: &mut Self::Class) {
+            // The layout manager determines how child widgets are laid out.
+            klass.set_layout_manager_type::<gtk::BoxLayout>();
+            klass.install_properties(&PROPERTIES);
+        }
+
+        fn new() -> Self {
+            // Here we set the default orientation.
+            Self {
+                first_label: RefCell::new(None),
+                second_label: RefCell::new(None),
+                orientation: RefCell::new(gtk::Orientation::Horizontal),
+            }
+        }
+    }
+
+    impl ObjectImpl for CustomOrientable {
+        fn constructed(&self, obj: &Self::Type) {
+            self.parent_constructed(obj);
+
+            // Create the children labels.
+            let first_label = gtk::Label::new(Some("Hello"));
+            let second_label = gtk::Label::new(Some("World!"));
+            let layout_manager = obj
+                .get_layout_manager()
+                .unwrap()
+                .downcast::<gtk::BoxLayout>()
+                .unwrap();
+            layout_manager.set_spacing(6);
+            first_label.set_parent(obj);
+            second_label.set_parent(obj);
+            self.first_label
+                .replace(Some(first_label.upcast::<gtk::Widget>()));
+            self.second_label
+                .replace(Some(second_label.upcast::<gtk::Widget>()));
+        }
+
+        fn dispose(&self, _obj: &Self::Type) {
+            // Child widgets need to be manually unparented in `dispose()`.
+            if let Some(child) = self.first_label.borrow_mut().take() {
+                child.unparent();
+            }
+
+            if let Some(child) = self.second_label.borrow_mut().take() {
+                child.unparent();
+            }
+        }
+
+        fn set_property(&self, obj: &Self::Type, id: usize, value: &glib::Value) {
+            let prop = &PROPERTIES[id];
+
+            match *prop {
+                glib::subclass::Property("orientation", ..) => {
+                    let orientation = value.get().unwrap().unwrap();
+                    self.orientation.replace(orientation);
+                    // We have to set the value in our layout manager as well.
```

Comment:
Why store it in both places? You could also just use the layout manager or not?

Reply:
Actually that's what gtkbox does, not sure why though..

Reply:
:man_shrugging: More state to keep in sync, I don't like it

Reply:
Ah, that's only because GtkBox exposes the "orientation" property, so we could technically just bind both properties here. In general you would just set the proper orientation only once without having to implement orientable at all. 

So this more of a copy of the GtkBox I guess